src/quacc/calculators/espresso/espresso.py

```diff
@@ -101,10 +110,55 @@ def read_results(self, directory: Path | str) -> dict[str, Any]:
         dict
             The results dictionnary
         """
+
         results = read(Path(directory) / self.outputname, binary=self.binary)
         if "energy" not in results:
             results["energy"] = None
         return results
+    
+    def _outdir_handler(self, parameters: dict[str, Any]) -> dict[str, Any]:
+        """
+        Function that handles the various outdir of espresso binaries.
+        If the user-supplied path are absolute, they are resolved and checked
+        against the scratch_dir and results_dir. If they are relative, they are
+        resolved against the scratch_dir and results_dir. If they are not in
+        either of those directories, they are ignored. This case is discouraged,
+        are encouraged to use quacc's scratch_dir and results_dir.
+
+        Parameters
+        ----------
+        **kwargs
+            User-supplied kwargs
+
+        Returns
+        -------
+        kwargs
+            The merged kwargs
+        """
```

Comment:
Should be `parameters` for the name of the parameters and `dict[str, Any]` for the name of the returns.